### ISSUE TITLE
yaml implicit resolver leak

### ISSUE THREAD

**Issue by xqliang** [yaml/pyyaml#67]
Every time when we call `yaml.dump`, it will call the `resolve` method:
```python
    def resolve(self, kind, value, implicit):
        if kind is ScalarNode and implicit[0]:
            if value == u'':
                resolvers = self.yaml_implicit_resolvers.get(u'', [])
            else:
                resolvers = self.yaml_implicit_resolvers.get(value[0], [])
            resolvers += self.yaml_implicit_resolvers.get(None, [])
```
As show above, the last line [resolvers += self.yaml_implicit_resolvers.get(None, [])](https://github.com/yaml/pyyaml/blob/master/lib/yaml/resolver.py#L149) in `resolve`  will change the `yaml_implicit_resolvers` inplace by accident, making `yaml_implicit_resolvers`  size become more and more bigger, the `yaml.dump`  will slow down, and finally use up 100% CPU.

reproduce:
```python
import re

import yaml


def test_implicit_resolver_leak():
    # Add any custom resolver
    tag, regexp = '!any_resolver', re.compile('AnyResolver')
    yaml.add_implicit_resolver(tag, regexp)

    none_resolvers = yaml.Dumper.yaml_implicit_resolvers.get(None, [])
    assert (tag, regexp) in none_resolvers

    old_f_resolvers = yaml.Dumper.yaml_implicit_resolvers.get('f', [])
    assert (tag, regexp) not in old_f_resolvers

    # Dump at least one ScalarNode
    yaml.dump(False)

    new_f_resolvers = yaml.Dumper.yaml_implicit_resolvers.get('f', [])
    assert (tag, regexp) in new_f_resolvers  # leak here
```

**Comment by perlpunk** [comment]
Thanks! I can reproduce it here.

**Comment by perlpunk** [comment]
This was fixed by #441 which was released in 5.4.
This should have been in
https://github.com/yaml/pyyaml/projects/5
I guess.

@ingydotnet I close this for you.
